executor/common_linux.h

```diff
@@ -2417,17 +2416,24 @@ static unsigned long fs_image_segment_check(unsigned long size, unsigned long ns
 		size = IMAGE_MAX_SIZE;
 	return size;
 }
-#endif
 
-#if SYZ_EXECUTOR || __NR_syz_read_part_table
-// syz_read_part_table(size intptr, nsegs len[segments], segments ptr[in, array[fs_image_segment]])
-static long syz_read_part_table(volatile unsigned long size, volatile unsigned long nsegs, volatile long segments)
+// Setup the loop device needed for mounting a filesystem image. Takes care of
+// creating and initializing the underlying file backing the loop device and
+// returns the fds to the file and device.
+// Returns 0 on success, -1 otherwise.
+static int setup_loop_device(long unsigned size, long unsigned nsegs, struct fs_image_segment* segs, const char* memfd_name, const char* loopname, int* _memfd, int* _loopfd)
 {
-	char loopname[64], linkname[64];
-	int loopfd, err = 0, res = -1;
-	unsigned long i, j;
-	size = fs_image_segment_check(size, nsegs, segments);
-	int memfd = syscall(sys_memfd_create, "syz_read_part_table", 0);
+	int err = 0;
+	int loopfd;
+	int memfd;
+
+	if (!loopname || !memfd_name || !_memfd || !_loopfd) {
+		debug("setup_loop_device: received a NULL pointer\n");
+		return -1;
+	}
+
+	size = fs_image_segment_check(size, nsegs, segs);
+	memfd = syscall(sys_memfd_create, memfd_name, 0);
```

Comment:
I don't think this memfd_name is important to preserve, it's not used for anything meaningful. It just used to be different because we had code duplicated. But now it does not worth the additional clutter in all reproducers (most of which will probably use only one of the syscalls anyway). So just use something like "syzkaller" or something.